### DISCUSSION TITLE
c++ SDK: check for input image size

### DISCUSSION THREAD

**Discussion by Kaeseknacker** [open-mmlab/mmdeploy#1866] [Q&A]
Hi, I work with this c++ SDK example for TRT-detectors:
```c++
#include <cstdlib>
#include <opencv2/opencv.hpp>
#include "mmdeploy/detector.hpp"
 
int main() {
  const char* device_name = "cuda";
  int device_id = 0;
  std::string model_path = "/home/spraul/Code/mmdeploy/mmdeploy_model/faster-rcnn";
  std::string image_path = "/home/spraul/Code/mmdeploy/mmdetection/demo/demo.jpg";
 
  // 1. load model
  mmdeploy::Model model(model_path.c_str());
  // 2. create predictor
  mmdeploy::Detector detector(model, mmdeploy::Device{device_name, device_id});
  // 3. read image
  cv::Mat img = cv::imread(image_path);
  // 4. inference
  auto dets = detector.Apply(img);
  // 5. deal with the result. Here we choose to visualize it
  for (int i = 0; i < static_cast<int>(dets.size()); ++i) {
    const auto& box = dets[i].bbox;
    fprintf(stdout, "box %d, left=%.2f, top=%.2f, right=%.2f, bottom=%.2f, label=%d, score=%.4f\n",
            i, box.left, box.top, box.right, box.bottom, dets[i].label_id, dets[i].score);
    if (dets[i].score < 0.3) {
      continue;
    }
    cv::rectangle(img, cv::Point{(int)box.left, (int)box.top},
                  cv::Point{(int)box.right, (int)box.bottom}, cv::Scalar{0, 255, 0});
  }
  cv::imwrite("output_detection.png", img);
  return 0;
}
```
I use a dynamic model to process different image sizes. Before input the image to the model I want to check if the image size is in the dynamic range of the model.
Is it possible to get the dynamic input range from the model via the SDK? I can't find any function for that.

Or as a workaround: What kind of exception is `Apply` throwing if the image isn't inside the input range? I can't find out either.
I only get this error, but I don't know how to catch it:
```
[2023-03-09 13:12:37.298] [mmdeploy] [error] [trt_net.cpp:28] TRTNet: 3: [executionContext.cpp::setBindingDimensions::947] Error Code 3: API Usage Error (Parameter check failed at: runtime/api/executionContext.cpp::setBindingDimensions::947, condition: profileMinDims.d[i] <= dimensions.d[i]. Supplied binding dimension [1,3,280,280] for bindings[0] exceed min ~ max range at index 2, maximum dimension in profile is 2176, minimum dimension in profile is 320, but supplied dimension is 280.
)
terminate called without an active exception
```

**Comment by irexyc** [comment]
@Kaeseknacker 
Hi, sorry for late reply, it's better ask a question [there](https://github.com/open-mmlab/mmdeploy/issues)

The model range is determined when convert the model, you can check the deploy config. You can also take a look at detail.json in model directory, it saved some backend information which have input shapes.

**Comment by Kaeseknacker** [comment]
Hi,
thanks for your reply. I checked github/issues, but I thought this is the right place for asking questions :)

I also found the image size in the detail.json file. But I actually wanted to avoid parsing this file myself. I assume mmdeploy parses this file anyway, because it is necessary so that the model can be loaded.
Would be nice if the SDK could provide this information.
